tfhe/src/high_level_api/integers/types/base.rs

```diff
@@ -1678,3 +1678,145 @@ where
         GenericInteger::<P>::new(ciphertext, self.id)
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::conformance::ParameterSetConformant;
+    use crate::core_crypto::prelude::UnsignedInteger;
+    use crate::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+    use crate::shortint::{CiphertextModulus, PBSOrder};
+    use crate::{generate_keys, set_server_key, ConfigBuilder, FheUint8};
+    use rand::{thread_rng, Rng};
+
+    type IndexedParameterAccessor<Ct, T> = dyn Fn(usize, &mut Ct) -> &mut T;
+
+    type IndexedParameterModifier<'a, Ct> = dyn Fn(usize, &mut Ct) + 'a;
+
+    fn change_parameters<Ct, T: UnsignedInteger>(
+        func: &IndexedParameterAccessor<Ct, T>,
+    ) -> Vec<Box<IndexedParameterModifier<'_, Ct>>> {
+        vec![
+            Box::new(|i, ct| *func(i, ct) = T::ZERO),
+            Box::new(|i, ct| *func(i, ct) = func(i, ct).wrapping_add(T::ONE)),
+            Box::new(|i, ct| *func(i, ct) = func(i, ct).wrapping_sub(T::ONE)),
+        ]
+    }
+
+    #[test]
+    fn test_invalid_generic_integer() {
+        type Ct = FheUint8;
+
+        let config = ConfigBuilder::default().build();
+
+        let (client_key, _server_key) = generate_keys(config);
+
+        let ct = FheUint8::try_encrypt(0_u64, &client_key).unwrap();
+
+        assert!(
+            ct.is_conformant(&RadixCiphertextConformanceParams::from_pbs_parameters(
+                PARAM_MESSAGE_2_CARRY_2_KS_PBS,
+                4
+            ))
+        );
+
+        let breaker_lists = vec![
```

Comment:
This could also be an array `let breaker_lists = [..];`

Reply:
Done